cpp/src/strings/padding.cu

```diff
@@ -146,58 +146,54 @@ std::unique_ptr<column> pad(
 
-//
-// Although zfill is identical to pad(width,'left','0') this implementation is a little
-// more optimized since it does not need to calculate the size of the fillchar and can
-// directly write it to the output buffer without extra logic for multi-byte UTF-8 chars.
-//
 std::unique_ptr<column> zfill(
-  strings_column_view const& strings,
+  strings_column_view const& input,
   size_type width,
   rmm::cuda_stream_view stream,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
 {
-  size_type strings_count = strings.size();
-  if (strings_count == 0) return make_empty_column(type_id::STRING);
+  if (input.is_empty()) return make_empty_column(type_id::STRING);
 
-  auto strings_column = column_device_view::create(strings.parent(), stream);
+  auto strings_column = column_device_view::create(input.parent(), stream);
   auto d_strings      = *strings_column;
 
-  // copy bitmask
-  rmm::device_buffer null_mask = cudf::detail::copy_bitmask(strings.parent(), stream, mr);
-
   // build offsets column
   auto offsets_transformer_itr = thrust::make_transform_iterator(
     thrust::make_counting_iterator<int32_t>(0),
     compute_pad_output_length_fn{d_strings, width, 1});  // fillchar is 1 byte
   auto offsets_column = make_offsets_child_column(
-    offsets_transformer_itr, offsets_transformer_itr + strings_count, stream, mr);
-  auto d_offsets = offsets_column->view().data<int32_t>();
+    offsets_transformer_itr, offsets_transformer_itr + input.size(), stream, mr);
+  auto const d_offsets = offsets_column->view().data<int32_t>();
 
   // build chars column
-  auto const bytes =
-    cudf::detail::get_value<int32_t>(offsets_column->view(), strings_count, stream);
+  auto const bytes = cudf::detail::get_value<int32_t>(offsets_column->view(), input.size(), stream);
   auto chars_column = strings::detail::create_chars_child_column(bytes, stream, mr);
   auto d_chars      = chars_column->mutable_view().data<char>();
 
   thrust::for_each_n(rmm::exec_policy(stream),
                      thrust::make_counting_iterator<cudf::size_type>(0),
-                     strings_count,
+                     input.size(),
                      [d_strings, width, d_offsets, d_chars] __device__(size_type idx) {
                        if (d_strings.is_null(idx)) return;
-                       string_view d_str = d_strings.element<string_view>(idx);
-                       auto length       = d_str.length();
-                       char* out_ptr     = d_chars + d_offsets[idx];
+                       auto d_str   = d_strings.element<string_view>(idx);
+                       auto length  = d_str.length();
+                       auto in_ptr  = d_str.data();
+                       auto out_ptr = d_chars + d_offsets[idx];
+                       // if the string starts with a sign, output the sign first
+                       if (!d_str.empty() && (*in_ptr == '-' || *in_ptr == '+')) {
+                         *out_ptr++ = *in_ptr++;
```

Comment:
I always feel uncomfortable with combining the unary ops like this. My memory is not good to remember the execution order of the `*` vs `++` operators and can't verify the correctness of the statement `*xxx++`. Can you add parenthesis like last time I asked, please?

Reply:
Same answer as before. Honestly I would find parenthesis confusing myself.